### PR TITLE
[Library] Added stacked and grouped horizontal bar charts.

### PR DESCRIPTION
**Pull request by dwnoble**
Added:

* stacked and grouped horizontal bar chart web components
* bar chart sorting
* configurable maximum number of places
* bar chart and line chart legend styling

Screenshots:

## Horizontal bar charts

horizontal-bar

## bug fix: now sorting by decreasing population by default
bug-fix-sort-descending

## Updated legend style
updated-legend-style

Locally tested  equity page, NL page

### CODE REVIEW

File: static/js/chart/draw.ts [hunk 1]
```diff
@@ -938,11 +1021,192 @@ function drawGroupBarChart(
...
+  formatNumberFn: (value: number, unit?: string) => string,
+  options?: {
+    apiRoot?: string;
+    stacked?: boolean;
+    style?: {
+      barHeight?: number;
+      yAxisMargin?: number;
+    };
+    unit?: string;
+  }
```
Comment (shifucun) [inline static/js/chart/draw.ts:1051]: define the option type in static/js/chart/types.ts ?

**Review by shifucun — COMMENTED** [review]
nice refactor!

**Review by juliawu — APPROVED** [review]
Thank you for the refactoring! Please respond to shifucun's comment before merging.

File: static/js/chart/draw.ts (hunk 1 above)
Comment (dwnoble) [reply static/js/chart/draw.ts]: done

**Comment by dwnoble** [comment]
done! thanks!

**Pull request merged**